### ISSUE TITLE
BUG | No matching export in "chroma-js" for import "Color" and "mix"

### ISSUE THREAD

**Issue by 1216892614** [redotvideo/revideo#228]
# Reproduction

1. `npm init @revideo@latest`
    - `✔ Project name … my-revideo-project`
    - `✔ Project path … my-revideo-project`
    - `✔ Choose a starter template › Minial, standalone Revideo project`
1. `cd ./my-revideo-project`
1. `pnpm i`
1. `npm run start`

# Deps

from `package.json`

```json
  "dependencies": {
    "@revideo/2d": "^0.5.2",
    "@revideo/core": "^0.5.2",
    "@revideo/ffmpeg": "^0.5.2",
    "@revideo/renderer": "^0.5.2",
    "@revideo/vite-plugin": "^0.5.2",
    "chroma-js": "^2.6.0"
  },
  "devDependencies": {
    "@revideo/cli": "^0.5.2",
    "@revideo/ui": "^0.5.2",
    "typescript": "^5.2.2",
    "vite": "^4.5"
  }
```

from `pnpm-lock.yaml`

```yaml
  '@revideo/core@0.5.2':
    dependencies:
      '@types/chroma-js': 2.4.4
      chroma-js: 2.6.0
      mp4-wasm: 1.0.6
```

# Environment

- win11 + wsl2
- Node.js v20.12.2
- npm 10.5.0
- pnpm 9.1.1

# Error logs

```sh
> npm run start                                                                                                                                                                                                                (base)

> my-revideo-project@0.0.0 start
> vite


  VITE v4.5.3  ready in 325 ms

  ➜  Local:   http://localhost:9000/
  ➜  Network: use --host to expose
  ➜  press h to show help
✘ [ERROR] No matching export in "node_modules/.pnpm/chroma-js@2.6.0/node_modules/chroma-js/index.js" for import "Color"

    node_modules/.pnpm/@revideo+core@0.5.2/node_modules/@revideo/core/lib/types/Color.js:1:9:
      1 │ import { Color, mix } from 'chroma-js';
        ╵          ~~~~~

✘ [ERROR] No matching export in "node_modules/.pnpm/chroma-js@2.6.0/node_modules/chroma-js/index.js" for import "mix"

    node_modules/.pnpm/@revideo+core@0.5.2/node_modules/@revideo/core/lib/types/Color.js:1:16:
      1 │ import { Color, mix } from 'chroma-js';
        ╵                 ~~~

/home/hndpp/Code/ts-playground/node_modules/.pnpm/esbuild@0.18.20/node_modules/esbuild/lib/main.js:1649
  let error = new Error(`${text}${summary}`);
              ^

Error: Build failed with 2 errors:
node_modules/.pnpm/@revideo+core@0.5.2/node_modules/@revideo/core/lib/types/Color.js:1:9: ERROR: No matching export in "node_modules/.pnpm/chroma-js@2.6.0/node_modules/chroma-js/index.js" for import "Color"
node_modules/.pnpm/@revideo+core@0.5.2/node_modules/@revideo/core/lib/types/Color.js:1:16: ERROR: No matching export in "node_modules/.pnpm/chroma-js@2.6.0/node_modules/chroma-js/index.js" for import "mix"
    at failureErrorWithLog (/home/hndpp/Code/ts-playground/node_modules/.pnpm/esbuild@0.18.20/node_modules/esbuild/lib/main.js:1649:15)
    at /home/hndpp/Code/ts-playground/node_modules/.pnpm/esbuild@0.18.20/node_modules/esbuild/lib/main.js:1058:25
    at /home/hndpp/Code/ts-playground/node_modules/.pnpm/esbuild@0.18.20/node_modules/esbuild/lib/main.js:1525:9
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errors: [
    {
      detail: undefined,
      id: '',
      location: {
        column: 9,
        file: 'node_modules/.pnpm/@revideo+core@0.5.2/node_modules/@revideo/core/lib/types/Color.js',
        length: 5,
        line: 1,
        lineText: "import { Color, mix } from 'chroma-js';",
        namespace: '',
        suggestion: ''
      },
      notes: [],
      pluginName: '',
      text: 'No matching export in "node_modules/.pnpm/chroma-js@2.6.0/node_modules/chroma-js/index.js" for import "Color"'
    },
    {
      detail: undefined,
      id: '',
      location: {
        column: 16,
        file: 'node_modules/.pnpm/@revideo+core@0.5.2/node_modules/@revideo/core/lib/types/Color.js',
        length: 3,
        line: 1,
        lineText: "import { Color, mix } from 'chroma-js';",
        namespace: '',
        suggestion: ''
      },
      notes: [],
      pluginName: '',
      text: 'No matching export in "node_modules/.pnpm/chroma-js@2.6.0/node_modules/chroma-js/index.js" for import "mix"'
    }
  ],
  warnings: []
}

Node.js v20.12.2
```

**Comment by justusmattern27** [comment]
hey, thanks for reporting this! This is caused by a recent chroma-js release (2.5.0) that has breaking changes which affect revideo. You'll have to downgrade to 2.4.x to fix the issue.

PR #226 fixes this by pinning the chroma-js version. We'll release a new version that includes this fix today

**Comment by 1216892614** [comment]
> hey, thanks for reporting this! This is caused by a recent chroma-js release (2.5.0) that has breaking changes which affect revideo. You'll have to downgrade to 2.4.x to fix the issue.
> 
> PR #226 fixes this by pinning the chroma-js version. We'll release a new version that includes this fix today

still not work. BTW, Im using bun this time.

## package.json

```json
  "dependencies": {
    "@revideo/2d": "^0.5.2",
    "@revideo/core": "^0.5.2",
    "@revideo/ffmpeg": "^0.5.2",
    "@revideo/renderer": "^0.5.2",
    "@revideo/vite-plugin": "^0.5.2",
    "chroma-js": "2.4.2"
  },
  "devDependencies": {
    "@revideo/ui": "^0.5.2",
    "@revideo/cli": "^0.5.2",
    "typescript": "^5.2.2",
    "vite": "^4.5"
  }
```

## Error logs

```sh
> bun run start                                                                                                                                                                                                                (base) 
$ vite

  VITE v4.5.3  ready in 348 ms

  ➜  Local:   http://localhost:9000/
  ➜  Network: use --host to expose
  ➜  press h to show help
✘ [ERROR] No matching export in "node_modules/@revideo/core/node_modules/chroma-js/index.js" for import "Color"

    node_modules/@revideo/core/lib/types/Color.js:1:9:
      1 │ import { Color, mix } from 'chroma-js';
        ╵          ~~~~~

✘ [ERROR] No matching export in "node_modules/@revideo/core/node_modules/chroma-js/index.js" for import "mix"

    node_modules/@revideo/core/lib/types/Color.js:1:16:
      1 │ import { Color, mix } from 'chroma-js';
        ╵                 ~~~

```

**Comment by hkonsti** [comment]
Thanks again for reporting! We just released 0.5.3, which should fix this issue. Please reopen in case you still face issues after upgrading.
